xdas/io/febus.py

```diff
@@ -1,27 +1,89 @@
 import h5py
 import numpy as np
+import warnings
 
 from ..core.dataarray import DataArray
 from ..core.routines import concatenate
 from ..virtual import VirtualSource
 
 
-def read(fname):
-    """Open a febus file into a xdas DataArray"""
+def read(fname, overlaps=None, offset=None):
+    """
+    Open a Febus file into a xdas DataArray object.
+
+    The Febus file format contains a 3D array which is a stack of 2D (time, distance)
+    chunks of data that overlaps with each other. The overlaps must be trimmed and the
+    chunks concatenated to form a seamless dataset. Each chunk is associated with a
+    timestamp that is located at a fixed offset from the beginning of the chunk.
+
+    Because of poor documentation of the evolution of the Febus file format, it is
+    recommended to manually specify the overlap and offset parameters. If not provided,
+    the function will attempt to determine the correct values at your own risk.
+
+    Parameters:
+    -----------
+    fname : str
+        The filename of the Febus file to read.
+    overlaps : tuple of int, optional
+        A tuple specifying the overlap in number of sample to trim on both side of each
+        chunk of the data. If as scalar is provided, the same overlap will be applied
+        to both sides. If not provided, the function will attempt to determine the
+        correct overlap at your own risk.
+    offset : int, optional
+        The location of the timestamp within each block given as the number of samples
+        from the beginning. If not provided, the function will attempt to determine the
+        correct offset at you own risk.
+
+    Returns:
+    --------
+    DataArray
+        A data array containing the data from the Febus file.
+
+    """
     with h5py.File(fname, "r") as file:
         (device_name,) = list(file.keys())
         source = file[device_name]["Source1"]
         times = np.asarray(source["time"])
         zone = source["Zone1"]
         (name,) = list(zone.keys())
         chunks = VirtualSource(zone[name])
-        delta = [zone.attrs["Spacing"][1] / 1000.0, zone.attrs["Spacing"][0]]
+        delta = (zone.attrs["Spacing"][1] / 1000.0, zone.attrs["Spacing"][0])
     name = "".join(["_" + c.lower() if c.isupper() else c for c in name]).lstrip("_")
-    noverlap = chunks.shape[1] // 4
-    chunks = chunks[:, noverlap:-noverlap, :]
-    times = times + noverlap * delta[0]
+
+    match overlaps:
+        case None:
+            warnings.warn(
+                "No overlap specified, Xdas will try its best to find the correct trimming"
+            )
+            if "BlockOverlap" in zone.attrs:
+                overlap = zone.attrs["BlockOverlap"][0]
+            else:
+                overlap = chunks.shape[1] // 4
```

Comment:
@ClaudioStrumia Check this please.

Reply:
case None raises an error beacouse zone can be accessed only when the h5 file is read, so either we open again the file here, or we save overlaps while the file is read.

Reply:
```suggestion
    match overlaps:
        case None:
            warnings.warn(
                "No overlap specified, Xdas will try its best to find the correct trimming"
            )
            with h5py.File(fname, "r") as file:
                (device_name,) = list(file.keys())
                source = file[device_name]["Source1"]
                times = np.asarray(source["time"])
                zone = source["Zone1"]
                if "BlockOverlap" in zone.attrs:
                    overlap = zone.attrs["BlockOverlap"][0]
                else:
                    overlap = chunks.shape[1] // 4
                overlaps = (overlap, overlap)
        case int(overlap):
            overlaps = (overlap, overlap)
        case (int(), int()):
            pass
        case _:
            raise ValueError("overlaps must be a integer or a tuple of two integers")
```

Reply:
Probably better **not** to reopen the file again. So either try to get block_overlap when the file is open OR put everything in the with statement.

Reply:
So in the with statement:

```python
block_overlap = zone.attrs["BlockOverlap"] if "BlockOverlap" in zone.attrs else None
```

Then

```python
if block_overlap is None:
    ...
else:
   ...
```

Reply:
Are you aware of other fields that could help ? You told be it can be in percents or in number of samples.

Reply:
Yes, the other problem is that the attribute zone.attrs["BlockOverlap"][0] is not the number of samples to trim, but the percentage of trimming with respect to one block. I was going to add a possible approach to identify it automatically by using zone.attrs["BlockOverlap"][0] which is the time duration of each chunk (in ms) and the dt (delta[0]). In this way one could obtain the numer of oversampled samples in each chunk and trim. In doing so, there should also be included the possibility to handle cases with no even number of oversample (e.g. 125Hz with 5% overlap).
The other way around is to put the default at 50% (trimming of n // 4) when case is None, and specify to the user that he must insert the values manually if the oversampling is different.

Reply:
Yeah we can also remove completely the automatic overlap / offset detection. It is maybe safer. You know more about me how likely will be an automatic detection method to get the correct choices. if it is lower than 75% then I would remove that, turn warnings into ValueError and force the user to put the values himself.